packages/wallets/src/components/WalletHistory.tsx

```diff
@@ -197,17 +202,26 @@ export default function WalletHistory(props: Props) {
     }
   );
   const { wallet, loading: isWalletLoading, unit } = useWallet(walletId);
+  const [transactions, setTransactions] = React.useState<Transaction[]>([]);
   const {
-    transactions,
+    transactions: transactionWithoutIncomingMemos,
     isLoading: isWalletTransactionsLoading,
     page,
     rowsPerPage,
     count,
     pageChange,
   } = useWalletTransactions(walletId, 10, 0, 'RELEVANCE');
+
+  React.useEffect(() => {
+    if (transactionWithoutIncomingMemos?.length) {
+      setTransactions(transactionWithoutIncomingMemos);
+    }
+  }, [transactionWithoutIncomingMemos]);
```

Comment:
Is this effect necessary? It's not obvious to me why we need this.

Reply:
Well, you might be right on this one, but I just didn't and still don't know how to put memo inside a rendered row other way.

The problem is that we need to mutate "transactions" inside "TableControlled" component each time we fetch a new memo and it is the "cols" variable that defines how each row will be rendered and the rendering rule is defined outside the the main component, so we can not access state from each row.

```
      <TableControlled
        cols={cols}
        rows={transactions ?? []}
```        

I was also tempted to create another hook for that in a separate file, but I would then need to put useEffect there, so I just did it this way with less code.